### ISSUE TITLE
Cannot open valid GLTFs that have animation channel targets without a node

### ISSUE THREAD

**Issue by carolhmj** [google/model-viewer#4153]
### Description

Hey there! A user brought this to our attention on the Babylon.js forums. I generated a similar case here: https://playground.babylonjs.com/#CBGEQX#1495, where we have 2 boxes, and one animation group targeting this first box. When exporting the scene the first box is filtered out, but the animation group referencing it is not. The resulting file has animation channels that don't reference any node:

```json
"animations": [
    {
      "name": "my group",
      "channels": [
        {
          "sampler": 0,
          "target": {
            "path": "scale"
          }
        },
        {
          "sampler": 1,
          "target": {
            "path": "rotation"
          }
        }
      ],
      "samplers": [
        {
          "interpolation": "LINEAR",
          "input": 3,
          "output": 4
        },
        {
          "interpolation": "LINEAR",
          "input": 5,
          "output": 6
        }
      ]
    }
  ],
```
which is considered a valid GLTF file: 
![image](https://user-images.githubusercontent.com/6002144/223806782-fec71061-aa96-40b3-9dbd-c61a4a83ef91.png)

and opens in the Babylon sandbox (https://sandbox.babylonjs.com/) but not on the model-viewer editor:

```javascript
space-opera.js:70437 
        
       TypeError: Cannot read properties of undefined (reading 'name')
    at GLTFParser._loadNodeShallow (space-opera.js:68848:28)
    at GLTFParser.loadNode (space-opera.js:68777:30)
    at space-opera.js:67436:34
    at GLTFParser._invokeOne (space-opera.js:67385:19)
    at GLTFParser.getDependency (space-opera.js:67434:24)
    at GLTFParser.loadAnimation (space-opera.js:68585:28)
    at space-opera.js:67488:39
    at GLTFParser._invokeOne (space-opera.js:67385:19)
    at GLTFParser.getDependency (space-opera.js:67486:24)
    at space-opera.js:67538:19
```

this is not a very common case and we're going to fix the GLTF export on the Babylon side so the animation groups referencing a filtered node are filtered out too, as the users probably won't have use for it anyway, but I thought we'd let y'all know anyway just in case 😅 

#### Live Demo
Load the file downloaded by https://playground.babylonjs.com/#CBGEQX#1495 in model-viewer editor

### Version
- model-viewer: v3.0.1 

### Browser Affected
Doesn't look browser specific but I tested on
- [x] Chrome, version: 110.0.5481.178 (Official Build) (64-bit) (cohort: Stable) 
- [x] Edge, version: 110.0.1587.63 (Official build) (64-bit) 
- [x] Firefox, version: 110.0.1
- [ ] IE
- [ ] Safari

### OS

- [ ] Android
- [ ] iOS
- [ ] Linux
- [ ] MacOS
- [x] Windows: Win11 10.0

**Comment by elalish** [comment]
Thanks for bringing it to our attention. Regardless I think this is actually a three.js GLTFLoader bug (unless the three.js editor handles this better?). I'm also curious if it's a validator/schema bug - does this file actually make sense? @donmccurdy what do you think?

**Comment by donmccurdy** [comment]
I believe this is allowed by the glTF spec in order to accommodate future extensions, animating properties that might not be associated with nodes. KHR_animation_pointer (draft) for example. It would of course be better for exporters to use that behavior only intentionally.

It can be fixed in GLTFLoader too, see https://github.com/mrdoob/three.js/pull/24770 and some related PRs, though I think it has been a low-priority issue unless/until there's an extension that uses this behavior.

**Comment by elalish** [comment]
Fixed by #4186
